### PR TITLE
Re-emit all events

### PR DESCRIPTION
**Pull request by mrudulpatil18**
Issue: #2 

Changes:
- Created a global WebWorkerEventEmitter to handle event re-emission
- Updated the `on` method in the web worker API 
- Provided a test that checks for specific render lifecycle 

Usage example:
```
circuitWebWorker.on(
    "renderable:renderLifecycle:PcbComponentRender:start",
    (eventData) => {
      console.log(eventData)
    },
  )
```

cc: @seveibar 

/claim #2

### CODE REVIEW

File: tests/file.test.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { createCircuitWebWorker } from "lib"
```
Comment (seveibar) [inline tests/file.test.tsx]: can you rename this test file

File: webworker/index.ts [hunk 2]
```diff
@@ -10,7 +10,32 @@ import * as React from "react"
...
+      try {
+        listener(...args)
+      } catch (error) {
+        console.error(`Error in event listener for ${event}:`, error)
+      }
+    })
+  }
+}
+
+const globalEventEmitter = new WebWorkerEventEmitter()
```
Comment (seveibar) [inline webworker/index.ts]: if this is just a normal event emitted (it looks like it is) you can just initialize `EventEmitter` from node

Comment (seveibar) [reply webworker/index.ts]: also does this have to be global? Why not just initialize an event emitter inside the `create*` method?

**Review by seveibar — COMMENTED** [review]
this is pretty good, but i'm just a bit confused about some details that seem like they could maybe be cleaned up? Can this be simplified a bit by using the builtin node event emitter?

**Comment by mrudulpatil18** [comment]
I didn't have experience working with EventEmitters etc, had to learn along the way.
The updated version does simplify everything thing soo much.

Note: 
> also does this have to be global? Why not just initialize an event emitter inside the `create*` method?

 - Removed the event re-emitting logic since we'll use the circuit's built-in event system directly                     
 - Update the on method to use the circuit's event system directly

**Review by seveibar — APPROVED** [review]
nice!

**Pull request merged**